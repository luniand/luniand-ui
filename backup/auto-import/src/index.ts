import * as UislandComponents from "@uisland-ui/vue-next"

export const componentResolver = (name: string) => {
  if (name in UislandComponents) {
    return {
      importName: name,
      path: `@uisland-ui/vue-next`,
    }
  }
}
