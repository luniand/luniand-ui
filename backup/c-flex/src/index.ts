export { default as CFlex } from "./flex"
