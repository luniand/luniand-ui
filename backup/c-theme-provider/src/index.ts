import {
  h,
  defineComponent,
  Fragment,
  PropType,
  provide,
  inject,
  computed,
} from "vue"
import { Theme } from "@uisland-ui/vue-theme"
import { ComponentWithProps } from "@uisland-ui/vue-system"

export interface CThemeProviderProps {
  value?: Theme
}

const CThemeProvider: ComponentWithProps<CThemeProviderProps> = defineComponent(
  {
    name: "CThemeProvider",
    props: {
      value: {
        type: [Object] as PropType<Theme>,
        default: () => undefined,
      },
    },
    setup(props, { slots }) {
      const pluginTheme = inject("$uislandTheme")
      const applicationTheme = computed(() => props.value || pluginTheme)
      provide("$uislandTheme", applicationTheme.value)
      return () => h(Fragment, slots.default?.({ $uislandTheme: props.value }))
    },
  }
)

export default CThemeProvider
