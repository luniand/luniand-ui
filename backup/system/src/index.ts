export * from "./uisland"
export * from "./providers"
export * from "./system.types"
export * from "./system.utils"
export * from "./composables/use-uisland"
export * from "./composables/use-style-config"
export * from "@uisland-ui/styled-system"
