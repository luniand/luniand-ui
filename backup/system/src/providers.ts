import { createContext } from "@uisland-ui/vue-utils"
import { Dict } from "@uisland-ui/utils"
import { SystemStyleObject } from "@uisland-ui/styled-system"
import { ComputedRef } from "@vue/reactivity"

const [StylesProvider, useStyles] = createContext<
  ComputedRef<Dict<SystemStyleObject> | SystemStyleObject>
>({
  name: "StylesContext",
  errorMessage:
    "useStyles: `styles` is undefined. Seems you forgot to provide `StylesProvider(...)` ",
})

export { StylesProvider, useStyles }

export const createStylesContext = (componentName: string) =>
  createContext<ComputedRef<Dict<SystemStyleObject> | SystemStyleObject>>({
    name: `${componentName}StylesContext`,
    errorMessage: `useStyles: "styles" is undefined. Seems you forgot to wrap the components in "<${componentName} />" `,
  })
