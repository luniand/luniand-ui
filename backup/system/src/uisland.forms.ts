import {
  PropType,
  SetupContext,
  computed,
  getCurrentInstance,
  ComponentInternalInstance,
} from "vue"
import { UnionStringArray } from "@uisland-ui/utils"

const inputTypes = [
  "button",
  "checkbox",
  "color",
  "date",
  "datetime-local",
  "email",
  "file",
  "hidden",
  "image",
  "month",
  "number",
  "password",
  "radio",
  "range",
  "reset",
  "search",
  "submit",
  "tel",
  "text",
  "time",
  "url",
  "week",
] as const

export type InputTypes = UnionStringArray<typeof inputTypes>

export interface FormElementProps {
  emits: string[]
  props: Record<string, unknown>
  handleValueChange: (
    props: any,
    type?: InputTypes
  ) => (emit: SetupContext["emit"]) => Record<string, unknown>
}

export const formElements: Record<string, FormElementProps> = {
  input: {
    emits: ["input", "change", "onUpdate:modelValue"],
    props: {
      modelValue: [Boolean, String] as PropType<boolean | string>,
    },
    handleValueChange(props: any, type?: InputTypes) {
      return (emit: SetupContext["emit"]) => {
        return {
          ...(type === "checkbox" && {
            checked: props.modelValue,
            // value: props.modelValue,
          }),
          onChange: (event: Event) => {
            if (type === "checkbox") {
              emit(
                "change",
                !(event?.target as HTMLInputElement).checked,
                event
              )
              emit(
                "update:modelValue",
                !(event?.target as HTMLInputElement).checked,
                event
              )
              return
            }
          },
          onInput: (event: any) => {
            emit(
              "input",
              (event?.currentTarget as HTMLInputElement).value,
              event
            )
            emit(
              "update:modelValue",
              (event?.currentTarget as HTMLInputElement).value,
              event
            )
          },
        }
      }
    },
  },
  textarea: {
    emits: ["input", "change", "onUpdate:modelValue"],
    props: {
      modelValue: [Boolean, String] as PropType<boolean | string>,
    },
    handleValueChange(props: any, type?: InputTypes) {
      return (emit: SetupContext["emit"]) => {
        return {
          onInput: (event: any) => {
            emit(
              "input",
              (event?.currentTarget as HTMLInputElement).value,
              event
            )
            emit(
              "update:modelValue",
              (event?.currentTarget as HTMLInputElement).value,
              event
            )
          },
        }
      }
    },
  },
  select: {
    emits: ["input", "change", "onUpdate:modelValue"],
    props: {
      modelValue: [Boolean, String] as PropType<boolean | string>,
    },
    handleValueChange(props: any, type?: InputTypes) {
      return (emit: SetupContext["emit"]) => {
        return {
          onChange: (event: any) => {
            emit(
              "input",
              (event?.currentTarget as HTMLInputElement).value,
              event
            )
            emit(
              "update:modelValue",
              (event?.currentTarget as HTMLInputElement).value,
              event
            )
          },
        }
      }
    },
  },
}

export type UislandFactoryElements = "input" | "select" | "textarea"

interface UislandFactoryElementHandlers {
  onChange?: Function[]
  onInput?: Function[]
}

const events = new Map<
  ComponentInternalInstance["uid"],
  UislandFactoryElementHandlers
>()

export function useFormElement(element: UislandFactoryElements, props: any) {
  const instance = getCurrentInstance()
  const uid = computed(() => instance?.uid || Date.now())

  const elProps = computed(() => ({
    checked: props.modelValue,
    value: props.moddelValue,
  }))

  return {
    elProps,
  }
}
