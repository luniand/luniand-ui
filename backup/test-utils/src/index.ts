export * from "./render"
export * from "./focus"
