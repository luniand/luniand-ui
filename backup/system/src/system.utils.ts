import { UnionStringArray, omit } from "@uisland-ui/utils"
import { keyframes, injectGlobal } from "@emotion/css"
import { PropType } from "vue"
import { ThemingProps } from "./system.types"

/**
 * Carefully selected html elements for uisland components.
 * This is mostly for `uisland.[element]` syntax.
 *
 * Adapted from React package
 */
export const domElements = [
  "a",
  "b",
  "article",
  "aside",
  "blockquote",
  "button",
  "caption",
  "cite",
  "circle",
  "code",
  "dd",
  "div",
  "dl",
  "dt",
  "fieldset",
  "figcaption",
  "figure",
  "footer",
  "form",
  "h1",
  "h2",
  "h3",
  "h4",
  "h5",
  "h6",
  "header",
  "hr",
  "iframe",
  "img",
  "input",
  "kbd",
  "label",
  "li",
  "mark",
  "main",
  "nav",
  "ol",
  "p",
  "path",
  "pre",
  "q",
  "rect",
  "s",
  "svg",
  "section",
  "select",
  "strong",
  "small",
  "span",
  "sub",
  "sup",
  "table",
  "tbody",
  "td",
  "textarea",
  "tfoot",
  "th",
  "thead",
  "tr",
  "ul",
] as const

export type DOMElements = UnionStringArray<typeof domElements>

export type DeepPartial<T> = {
  [P in keyof T]?: DeepPartial<T[P]>
}

export type ToPropType<T> = {
  [P in keyof T]?: PropType<T[P]>
}

export { keyframes, injectGlobal }
