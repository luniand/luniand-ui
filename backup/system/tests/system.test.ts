import { render, screen } from "../../test-utils/src"
import { defineComponent, h } from "vue"
import { uisland } from "../src"

describe("uisland() works", () => {
  const renderComponent = (options?: Record<string, any>) =>
    render({
      components: {
        uisland: defineComponent({
          setup(_, { slots }) {
            return () => h(uisland("span", {}), {}, slots)
          },
        }),
      },
      template: `<uisland>child-element</uisland>`,
      ...options,
    })

  it("should be render properly", () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it("should be render default slot", () => {
    renderComponent()
    expect(screen.getByText("child-element")).toBeInTheDocument()
  })
})

describe("as prop", () => {
  const FakeRouterLink = defineComponent({
    props: ["to"],
    setup(props, { slots }) {
      return () => h("a", { href: props.to }, { default: () => slots })
    },
  })
  it('render dom elements like as="h1"', () => {
    const { container, asFragment } = render({
      components: {
        uisland: defineComponent({
          setup(_, { slots }) {
            return () => h(uisland("h1", {}), {}, slots)
          },
        }),
      },
      template: `<uisland>as h1</uisland>`,
    })

    expect(asFragment()).toMatchSnapshot()
    expect(container.querySelector("h1")).toBeInTheDocument()
  })

  it('resolve global vue components like as="router-link"', () => {
    const { container, asFragment } = render(
      {
        components: {
          uisland: defineComponent({
            props: ["as"],
            setup(props, { slots, attrs }) {
              return () => h(uisland(props.as), {}, slots)
            },
          }),
        },
        template: `<uisland as="fake-router-link" to="https://google.com">as router-link</uisland>`,
      },
      // testing-library options
      {
        global: {
          components: {
            FakeRouterLink,
          },
        },
      }
    )

    expect(container.querySelector("a")).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('render Vue Component like :as="RouterLink"', () => {
    const { container, asFragment } = render({
      components: {
        uisland: defineComponent({
          setup(_, { slots }) {
            return () =>
              h(
                uisland(FakeRouterLink, { to: "https://vue.uisland-ui.com/" }),
                {},
                slots
              )
          },
        }),
      },
      template: `<uisland>as RouterLink</uisland>`,
    })

    expect(container.querySelector("a")).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
