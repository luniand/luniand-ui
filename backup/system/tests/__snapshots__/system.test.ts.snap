// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as prop render Vue Component like :as="RouterLink" 1`] = `
<DocumentFragment>
  <a
    class="css-0"
    href="https://vue.uisland-ui.com/"
  >
    as RouterLink
  </a>
</DocumentFragment>
`;

exports[`as prop render dom elements like as="h1" 1`] = `
<DocumentFragment>
  <h1
    class="css-0"
  >
    as h1
  </h1>
</DocumentFragment>
`;

exports[`as prop resolve global vue components like as="router-link" 1`] = `
<DocumentFragment>
  <a
    class="css-0"
    href="https://google.com"
  >
    as router-link
  </a>
</DocumentFragment>
`;

exports[`uisland() works should be render properly 1`] = `
<DocumentFragment>
  <span
    class="css-0"
  >
    child-element
  </span>
</DocumentFragment>
`;
